api/v1alpha1/etcdcluster_types.go

```diff
@@ -50,10 +53,56 @@ type ProviderConfig struct {
 	CertManagerCfg *ProviderCertManagerConfig `json:"certManagerCfg,omitempty"`
 }
 
+type AltNames struct {
+	// DNSNames accepts a string array, if empty programmatically populates the
+	// CommonName as DNSNames
+	// +optional
+	DNSNames []string `json:"dnsNames,omitempty"`
+
+	// IPs accepts a net.IP array of IP Addresses as alternative names
+	// +optional
+	IPs []net.IP `json:"ipAddresses,omitempty"`
+}
+
+type CommonConfig struct {
+	// CommonName accepts a string for a human-readable name of the certificate
+	// +optional
+	CommonName string `json:"commonName,omitempty"`
+
+	// Organization accepts a string array to be used on the Certificate.
+	// +optional
+	Organization []string `json:"organizations,omitempty"`
+
+	// AltNames contains the domain names and IP addresses that will be added
+	// to the x509 certificate SubAltNames fields. The values will be passed
+	// directly to the x509.Certificate object.
+	AltNames AltNames `json:"altNames,omitempty"`
+
+	// ValidityDuration accepts time.Duration, if empty defaults to 90 days
+	// +optional
+	ValidityDuration time.Duration `json:"validityDuration,omitempty"`
```

Comment:
If you look at the generated yaml file `config/crd/bases/operator.etcd.io_etcdclusters.yaml`, the type is converted to `int64` (nanaseconds). It will be hard for users to configure it. We expect that users should be able configure human readable values something like "90d", "48h", "100d12h" etc

Reply:
In that case, it will be better to accept it as a string and parse it internally when setting the internal config.